src/frontend/src/components/AddUser/index.tsx

```diff
@@ -0,0 +1,261 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { AppState } from '../../redux/store';
+import { UserState } from '../../redux/users/types';
+import validateEmail from '../../service/email-validation';
+import Logo from '../Logo';
+
+interface Props {
+  users: UserState;
+}
+
+interface State {
+  email: string;
+  password: string;
+  name: string;
+  group: string;
+  role: string;
+  invalidCredentials: boolean;
+  invalidResponse: boolean;
+  invalidEmail: boolean;
+  missingPassword: boolean;
+}
+
+class AddUser extends React.Component<Props, State> {
+  public state: State = {
+    email: '',
+    password: '',
+    name: '',
+    group: '',
+    role: '',
+    invalidCredentials: false,
+    invalidResponse: false,
+    invalidEmail: false,
+    missingPassword: false
+  };
+
+  public handleChange = (e: React.BaseSyntheticEvent) => {
+    // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/26635 for why we're
+    // using the "any" type.
+    console.log(e);
+    this.setState<any>({
+      [e.target.id]: e.target.value
+    });
+    console.log(this.state);
+  };
+
+  public handleSubmit = (event: React.BaseSyntheticEvent) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    // Standardize email-lowerCase and trim both form inputs
+    this.setState(
+      {
+        email: this.state.email.toLowerCase().trim(),
+        password: this.state.password.trim()
+      },
+      () => {

```

Comment:
It is probably good to add a 

// TODO: Submit to backend

so that until that is implemented anyone who reads this code isn't confused as to why there is an empty arrow function here.